### PR TITLE
bcda-764: retrofit router.go to use new auth provider

### PR DESCRIPTION
**Pull request by alexhova**
### Fixes [BCDA-764](https://jira.cms.gov/browse/BCDA-764)
middleware.go and therefore router.go uses the old backend implementation of auth.  it should use the new provider implementation.

### Proposed changes:
- move the alpha plugin into the auth package
- move GetAuthProvider() to auth package
- simplify auth middleware to use provider and router to use new middleware
- remove AllClaims type
- fix tests

### Security Implications
this introduces some changes to how requests are authorized and validated.  i don't believe this introduced any security concerns but it does touch a vital area security-wise

### Acceptance Validation
tests passed locally.  let's see what travis says

### CODE REVIEW

File: bcda/auth/provider.go [hunk 1]
```diff
@@ -17,3 +21,13 @@ type Provider interface {
 	ValidateJWT(tokenString string) error
 	DecodeJWT(tokenString string) (jwt.Token, error)
 }
+
+func GetAuthProvider() Provider {
```
Comment (embh) [inline bcda/auth/provider.go]: With this being in `auth` now, could we call it `GetProvider()`?

File: bcda/router.go [hunk 2]
```diff
@@ -49,7 +48,7 @@ func NewDataRouter() http.Handler {
 	r := chi.NewRouter()
 	m := monitoring.GetMonitor()
 	r.Use(ConnectionClose)
-	r.With(auth.ParseToken, logging.NewStructuredLogger(), auth.RequireTokenAuth,
```
Comment (embh) [inline bcda/router.go]: We need the token in the context before logging. Without it, `sub`, `aco`, and `token_id` are missing from the request log entries (vs. master, for instance)

Comment (alexhova) [reply bcda/router.go]: nice catch!  I can't figure out a good way to test it though...

File: bcda/auth/provider.go (hunk 1 above)
Comment (alexhova) [reply bcda/auth/provider.go]: yep!

File: bcda/router.go (hunk 2 above)
Comment (DeirdreHolub) [reply bcda/router.go]: https://github.com/sirupsen/logrus#testing

Comment (alexhova) [reply bcda/router.go]: emily created a [ticket](https://jira.cms.gov/browse/BCDA-847)!

Comment (alexhova) [reply bcda/router.go]: and i can confirm that locally the token context is being logged:
```
api_1            | {"aco":"0c527d2e-2e8a-4808-b11d-0fa06baf8254","http_method":"GET","http_proto":"HTTP/1.1","http_scheme":"http","level":"info","msg":"request started","remote_addr":"172.21.0.1:38406","sub":"6baf8254-2e8a-4808-b11d-0fa00c527d2e","time":"2019-02-01T20:41:13Z","token_id":"611be673-834a-4069-b4fe-3643a3900508","ts":"Fri, 01 Feb 2019 20:41:13 UTC","uri":"http://localhost:3000/data/100/0c527d2e-2e8a-4808-b11d-0fa06baf8254.ndjson","user_agent":"PostmanRuntime/7.6.0"}
```

**Pull request merged**